OpenRA.Mods.Common/Effects/LineImpactProjectile.cs

```diff
@@ -0,0 +1,256 @@
+		public readonly WDist[] Speed = { new WDist(17) };
+
+		[Desc("The number of ticks between the beam", "causing warhead impacts in its area of effect.")]
+		public readonly int TicksBetweenImpacts = 3;
+
+		[Desc("How far beyond the target the", "projectile keeps on travelling.")]
+		public readonly WDist BeyondTargetRange = new WDist(128);
+
+		[Desc("How close to the target must the", "projectile be to apply weapon effects.")]
+		public readonly WDist MaxTargetRange = new WDist(128);
+
+		[Desc("The maximum reduction to warhead effects", "100 = 100% falloff = 0% damage.")]
+		public readonly int FalloffMaxReduction = 75;
+
+		[Desc("The distance before the falloff starts.")]
+		public readonly WDist FalloffStartDistance = new WDist(128);
+
+		[Desc("The distance before the falloff ends.")]
+		public readonly WDist FalloffEndDistance = new WDist(4096);
+
+		[Desc("Maximum offset at the maximum range.")]
+		public readonly WDist Inaccuracy = WDist.Zero;
+
+		[Desc("Can this projectile be blocked when hitting actors", "with an IBlocksProjectiles trait.")]
+		public readonly bool Blockable = false;
+
+		[Desc("The width of the beam, determining", "which actors are within the area of effect.")]
+		public readonly WDist BeamWidth = new WDist(512);
+
+		[Desc("Extra search radius beyond beam width. Required to ensure affecting actors with large health radius.")]
+		public readonly WDist TargetExtraSearchRadius = new WDist(2048);
+
+		[Desc("The duration of the beam projectile", "effectively determines the length of the beam.")]
+		public readonly int BeamFireDuration = 10;
+
+		[Desc("The visual width of the beam projectile", "for rendering only.")]
+		public readonly int BeamRenderWidth = 2;
+
+		[Desc("Beam color in (A,)R,G,B.")]
+		public readonly Color Color = Color.Red;
+
+		[Desc("Beam color is the player's color.")]
+		public readonly bool UsePlayerColor = false;
+
+		public IEffect Create(ProjectileArgs args)
+		{
+			var c = UsePlayerColor ? args.SourceActor.Owner.Color.RGB : Color;
+			return new LineImpactProjectile(this, args, c);
+		}
+	}
+
+	public class LineImpactProjectile : IEffect, ISync
+	{
+		readonly LineImpactProjectileInfo info;
+		readonly ProjectileArgs args;
+		[Sync] readonly WDist speed;
+
+		Color color;
+		WAngle angle;
+		WVec offSet;
+
+		[Sync] WPos headPos;
+		[Sync] WPos tailPos;
+		[Sync] WPos target;
+		int length;
+		int facing;
+		[Sync] int headTicks;
+		[Sync] int tailTicks;
+		bool headTravelling, tailTravelling;
+
+		[Sync] public Actor SourceActor { get { return args.SourceActor; } }
+
+		public LineImpactProjectile(LineImpactProjectileInfo info, ProjectileArgs args, Color color)
+		{
+			this.info = info;
+			this.args = args;
+			this.color = color;
+
+			// Both the head and tail start at the source actor, but initially only the head is travelling.
+			headPos = args.Source;
+			tailPos = headPos;
+			headTravelling = true;
+			tailTravelling = false;
+			angle = new WAngle();
```

Comment:
You don't need to initialize this.  It already has a default value of `WAngle.Zero`.